rust/src/lib/nm/conn_matcher.rs

```diff
@@ -237,15 +237,27 @@ impl NmConnectionMatcher {
         &self,
         base_iface: &BaseInterface,
     ) -> Vec<&NmConnection> {
-        let mut ret: Vec<&NmConnection> = if let Some(nm_conns) =
-            self.saved_by_name_and_type.get(&(
-                base_iface.name.to_string(),
-                NmIfaceType::from(&base_iface.iface_type),
-            )) {
-            nm_conns.iter().map(Rc::as_ref).collect()
-        } else {
-            Vec::new()
-        };
+        let mut ret: Vec<&NmConnection> = Vec::new();
+        // If interface type is unknown, all profiles with desired interface
+        // name will be returned.
+        if base_iface.iface_type == InterfaceType::Unknown {
+            for nm_conns in self.saved_by_name_and_type.iter().filter_map(
+                |((name, _), nm_conns)| {
+                    if name == base_iface.name.as_str() {
+                        Some(nm_conns)
+                    } else {
+                        None
+                    }
+                },
+            ) {
+                ret.extend(nm_conns.iter().map(Rc::as_ref));
+            }
```

Comment:
IMO it would make sense to add a new index `saved_by_name` at creation time, and use it here.